### ISSUE TITLE
Uncaught TypeError: _this.convert is not a function

### ISSUE THREAD

**Issue by bitkill** [dimitrinicolas/midi-controller-css#3]
[Enter steps to reproduce:]

1. Downloaded package
2. Tried running midi:convert
3. Searched the code, where do I find convert?

**Atom**: 1.35.1 x64
**Electron**: 2.0.18
**OS**: linux 5.0.4-200.fc29.x86_64
**Thrown From**: [midi-controller-css](https://github.com/dimitrinicolas/midi-controller-css) package 1.2.0

### Stack Trace

Uncaught TypeError: _this.convert is not a function

```
At /home/rui/.var/app/io.atom.Atom/data/packages/midi-controller-css/lib/index.js:22

TypeError: _this.convert is not a function
    at HTMLElement.midiConvert (/packages/midi-controller-css/lib/index.js:22:40)
    at CommandRegistry.handleCommandEvent (/app/share/atom/resources/app/static/<embedded>:11:350055)
    at Object.didConfirmSelection (/app/share/atom/resources/app/static/<embedded>:11:692962)
    at SelectListView.confirmSelection (/app/share/atom/resources/app/static/<embedded>:11:858354)
    at HTMLDivElement.core:confirm (/app/share/atom/resources/app/static/<embedded>:11:853233)
    at CommandRegistry.handleCommandEvent (/app/share/atom/resources/app/static/<embedded>:11:350055)
    at KeymapManager.t.exports.KeymapManager.dispatchCommandEvent (/app/share/atom/resources/app/static/<embedded>:11:1230796)
    at KeymapManager.t.exports.KeymapManager.handleKeyboardEvent (/app/share/atom/resources/app/static/<embedded>:11:1226930)
    at WindowEventHandler.handleDocumentKeyEvent (/app/share/atom/resources/app/static/<embedded>:11:285234)
```

### Commands

```
     -2:07.8.0 fuzzy-finder:toggle-file-finder (div.package-detail)
     -2:06.8.0 editor:consolidate-selections (input.hidden-input)
     -2:06.8.0 core:cancel (input.hidden-input)
     -2:06.0 settings-view:open (div.package-detail)
     -2:04.9.0 key-binding-resolver:toggle (div.package-detail)
     -2:02.3.0 core:cancel (div.package-detail)
     -2:00.6.0 settings-view:open (div.package-detail)
     -1:59.1.0 fuzzy-finder:toggle-file-finder (div.package-detail)
  2x -1:56.8.0 core:backspace (input.hidden-input)
     -1:55 editor:consolidate-selections (input.hidden-input)
     -1:55 core:cancel (input.hidden-input)
  6x -1:36.6.0 core:backspace (input.hidden-input)
     -1:16.2.0 window:toggle-menu-bar (div.panels-item)
     -1:09.0 command-palette:toggle (div.panels-item)
     -1:06.5.0 core:confirm (input.hidden-input)
     -1:06.5.0 midi:convert (div.panels-item)
```

### Non-Core Packages

```
midi-controller-css 1.2.0 
```

**Comment by dimitrinicolas** [comment]
Hi!

Thank you for your precise report. It's been a long time since I've used this plugin. I reread my code and I think that this command `midi:convert` is totally useless, it is an error if this command remained in the code.

I unfortunately wrote this plugin very quickly, there is no configuration, informations message or visible command. Its usage is very basic (if it still works today): it edits CSS files when the editor cursor is inside a CSS selector block.

By example, create this CSS file:

```css
div {
  /* <- Place cursor here */
}
```

And then try some buttons, sliders and potentiometer on the Midi controller.

Also, this plugin only works with a Korg NanoKontrol 2, make sure that it is well recognized by your computer.

Dimitri

**Comment by bitkill** [comment]
Thanks for the reply, I have a nanoKontrol2, but am unable to connect it to the atom instance.
However, I wrote a simple program to connect it to a chrome page and it works.
I can read values from the browser, and display them. I viewed some of your code as guidance, and it was very helpful. 

However, I have a problem, when I connect this, no state is sent by default, I read the https://github.com/dimitrinicolas/midi-controller-css/blob/master/lib/index.js#L36
and it appears to send a sequence of data to set some kind of state, my understanding of this is to set a "standard" mode to receive data.
I am also looking for a way to set lights on the controller.
Can you point to some resources on this? If not, this was already of great help. Thanks!

**Comment by dimitrinicolas** [comment]
I stopped using this package because I started using VS Code approximately a year ago. I just quickly tried it and unfortunately it is not working anymore for me either. For your information, it was ultimately not that useful for writing CSS quickly. This project  was more of a technical challenge and curiosity.

You make me want to get back on this project! I'll try to find the actual issue soon.

I remember that the use of LEDs was very little documented by Korga

**Comment by dimitrinicolas** [comment]
I stopped using this package because I started using VS Code approximately a year ago. I just quickly tried it and unfortunately it is not working anymore for me either. For your information, it was ultimately not that useful for writing CSS quickly. This project  was more of a technical challenge and curiosity.

You make me want to get back on this project! I'll try to find the actual issue soon.

I remember that the use of LEDs was very little documented by Korg. The lines of code of this package that you referenced to is a bonus Led animation when the Midi controller is plugged in.

For using Leds, I helped myself with this code: https://github.com/overtone/overtone/blob/master/src/overtone/device/midi/nanoKONTROL2.clj#L133 and the obscure Midi implementation from Korg: https://www.korg.com/tmp/support/download/index.php/manual/0/159/2710/?country=us&page=manual&prodtype=0&prodid=159&fileid=2710. The Korg document help to understand the "magical sysex incantation". It says to the Korg NanoKontrol how you will want to control Leds states.

Finaly this for loop: https://github.com/dimitrinicolas/midi-controller-css/blob/master/lib/index.js#L75 is creating a basic animation with the controller Leds.

**Comment by dimitrinicolas** [comment]
I tried it again, and it is definitely working well.

The LEDs animation appears when opening Atom window and not when plugging in the Midi Controller to the computer. Plus: this plugin is only working when the Midi Controller is plugged into the computer before opening the Atom application.

I am using Atom 1.31.1, Electron 2.0.7, Node v8.9.3 on MacOS 10.14.1.

I then updated Atom to 1.35.1, Electron 2.0.18 and Node v8.9.3, the LEDs animation and the Midi Controller usage is still working.

I tried the forgotten wrong command `midi:convert` and the plugin keep working after that the error appears.
